pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/DataType.java

```diff
@@ -64,26 +62,27 @@ public enum DataType {
     public final boolean requiresEscaping;
 
     /**
-     * The set of {@link BasicType}s that map to this type. Multiple types can map to a single instance of this enum.
+     * The set of primitive type names that map to this type. Multiple types can map to a single instance of this enum.
+     * Note: these strings are not case-normalized.
      */
-    private final Set<BasicType> basicTypes;
+    private final Set<String> basicTypeNames;
 
     /**
-     * A case insensitive map of the enum name to its instance. The case metadata is not guaranteed to have the correct
+     * A map of the lower-case-normalized enum name to its instance. The case metadata is not guaranteed to have the correct
      * case.
      */
-    private static final Map<String, DataType> CASE_INSENSITIVE_MAP = new HashMap<>();
+    private static final Map<String, DataType> CASE_NORMALIZED_MAP = new HashMap<>();
```

Comment:
I renamed this since it's not actually case-insensitive (e.g. with equality and hash functions that don't depend on case).